src/EthPool.sol

```diff
@@ -59,21 +55,26 @@ contract EthPool is Pool {
         uint64 lastUpdate,
         uint128 lastBalance,
         Receiver[] calldata currReceivers,
-        uint128 withdraw,
+        uint128 reduceBalance,
         Receiver[] calldata newReceivers
-    ) public payable returns (uint128 newBalance, uint128 withdrawn) {
+    ) public payable returns (uint128 newBalance, int128 realBalanceDelta) {
         return
             _updateSender(
                 _senderId(msg.sender, subSenderId),
                 lastUpdate,
                 lastBalance,
                 currReceivers,
-                uint128(msg.value),
-                withdraw,
+                _balanceDelta(reduceBalance),
                 newReceivers
             );
     }
 
+    function _balanceDelta(uint128 reduceBalance) internal view returns (int128 balanceDelta) {
```

Comment:
Implementation seems correct. However with the current solution if I send 1 ETH and I want to reduce 2 ETH. We would withdraw effectively 1 ETH. 
However, we could also say we `reduceBalance` is more dominate and we want a `msg.value `of 0. If you however still send something, we would just return it.  Which means we would send 3 ETH back in total.

Reply:
The current solution is identical to what was before, the question is if it's a good approach? The current API is simple and does what you tell it even if it's stupid, there are no special cases.

Your proposal makes `reduceBalance` work more like `balanceDelta`, but it adds a special case for `reduceBalance == 0` and makes the parameters influence each other. To me that's a bit more complex.

Reply:
> The current solution is identical to what was before, the question is if it's a good approach? 
Yes. I agree it is also not ideal. The other alternative would be to just revert in that case.

Reply:
I've been thinking about it and I think that you're right, we should revert. There's no sane use case for non-zero both `reduceBalance` and `msg.value`, it's almost certainly a user's mistake. There's also no obvious behavior in that case, everything other than a revert is potentially a costly surprise.

Reply:
Yeah, It is most likely a mistake and a revert reduces the complexity for us. A case less we need to handle correctly.

Reply:
Fixed